### DISCUSSION TITLE
(completion-at-point) brings up menu, but I can't do any with it

### DISCUSSION THREAD

**Discussion by SequentialDesign** [minad/corfu#381] [Q&A]
when I call (completion-at-point) interactively, I get a menu, but I can't do anything with it.  

![corfu 1](https://github.com/minad/corfu/assets/82598112/92ccd0e7-fb67-46b7-ba57-0b3afa33cee8)

the menu is stuck and doesn't do anything.  the same thing happened with my real init (the one in the pic is super minimal (just for testing.  the only package I have on it is corfu)).  I want to call `completion-at-point` in a function, but I can't get it to work.  

I'm using windows 10. I have not tested it on linux yet.  have you seen something like this before ?

I tested it on wsl, and the result is the same: `completion-at-point` gives me a useless menu 

![corfu 2](https://github.com/minad/corfu/assets/82598112/2201f711-73cf-406d-89e6-b479ea817304)

**Comment by kartikynwa** [comment]
> I have not tested it on linux yet. have you seen something like this before ?

Happens on Linux too. This does not happen if the popup appears automatically due to `corfu-auto` but following the steps you mention I encounter the exact same problem.

**Comment by minad** [comment]
Hi, I've seen such "useless popups" before, but not for a long time. Iirc these were caused by some problems with input/window focus of the child frame, but these issues have been fixed. I cannot reproduce this on my current installation with emacs -Q (Emacs 29.1 X11/Lucid). I enter and evaluate the following in the scratch buffer and Corfu works as expected:

~~~ elisp
(package-initialize)
(global-corfu-mode)
(def<M-TAB>
~~~

I cannot test on Windows but your report is also the first one about problems there. Please try to create a minimal recipe ideally under Linux. Which Emacs version, window manager (kde, gnome, i3, xmonad, ...) and window toolkit do you use (X11+gtk, X11+lucid, X11+notoolkit, X11+pgtk, Wayland+pgtk, ...)? It could be that this issue occurs only under certain special configurations with certain window managers.´

**Reply by kartikynwa** [reply]
What is <M-TAB> bound to here? `completion-at-point`?

**Reply by minad** [reply]
The answer is a `C-h k` away :)

**Reply by kartikynwa** [reply]
No sorry. I meant if *you* pressing M-TAB is invoking `completion-at-point` *for you* or not. Doing `C-h k` will tell me what it does for me but not what is going on in your example.

I am trying to create a minimal reproducible example for this btw.

**Comment by kartikynwa** [comment]
I used this config to reproduce the problem:

```elisp
(use-package corfu
  :ensure t
  :init
  (global-corfu-mode))

(use-package cape
  :ensure t
  :init
  (add-to-list 'completion-at-point-functions #'cape-dabbrev)
  (add-to-list 'completion-at-point-functions #'cape-file))

(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(package-selected-packages '(corfu)))
(custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 )
```

<details>

<summary>Version and system details</summary>

- emacs version: `GNU Emacs 29.1`
- corfu version: `0.38`
- cape version: `0.17`
- Emacs is using the GTK3 frontend on X, xorg-server version is `21.1.8`
- Distro is Void LInux. `uname -a` output: `Linux hostname 6.5.7_1 #1 SMP PREEMPT_DYNAMIC Wed Oct 11 08:20:04 UTC 2023 x86_64 GNU/Linux`

</details>

This problem only occurs if completion-at-point is called interactively (`M-x completion-at-point RET`). If I invoke it by pressing `M-TAB`, everything works perfectly.

**Reply by minad** [reply]
Thanks! That's a good one! I see the issue now. The clue is the `M-x completion-at-point RET` which is needed to reproduce this.

**Reply by minad** [reply]
I pushed a fix in https://github.com/minad/corfu/commit/b2b9a2312f58117514724e729fda43efc4433ecd. Please give it a check.

**Reply by kartikynwa** [reply]
I just tried it briefly in the scratch buffer and it solves the problem. I tried my best but could not cause it. Thanks for fixing it so fast.

**Comment by SequentialDesign** [comment]
OMG! thank you so much to both of you !! kartikynwa, thank you so much for explaining it and helping to get to the bottom of the issue !

and, minad, thanks for your great packages !

y'all have made my day !!

**Reply by kartikynwa** [reply]
I hope your day gets even better
